hoot-services/src/main/java/hoot/services/controllers/export/ExportResource.java

```diff
@@ -152,16 +156,37 @@ public Response export(ExportParams params, @Context HttpServletRequest request,
             List<Command> workflow = new LinkedList<>();
 
             if (inputType.equalsIgnoreCase("folder")) {
-                Long folder_id = Long.parseLong(params.getInput());
                 params.setInputType("db"); // make folder input really a db input...
-
-                for (Tuple mapInfo: FolderResource.getFolderMaps(user, folder_id)) { // get all maps in folder...
-                    params.setInput(Long.toString(mapInfo.get(maps.id)));
-                    params.setOutputName(mapInfo.get(maps.displayName));
-                    workflow.add(getCommand(user, jobId, params, debugLevel));
+                Long rootfolder = Long.parseLong(params.getInput());
+
+                // create stack with the folder id and output path as values starting with main export folder
+                // Having the output path is helpful for making the child directories since you can just take parent path
+                // and append child directory name to it
+                Stack<Map.Entry<Long, String>> folderList = new Stack<Map.Entry<Long, String>>(){{
```

Comment:
this is pretty much DFS algorithm. I needed the 2nd value because it makes it really easy to figure out child file output paths